### PR TITLE
Builds with v4.3.0 instead of fork of older version of SQL Cipher [ci full]

### PR DESCRIPTION
**Pull request by dp-mason**
### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - `automation/all_tests.sh` runs to completion and produces no failures
  - Note: For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry in [CHANGES_UNRELEASED.md](../CHANGES_UNRELEASED.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/master/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

### CODE REVIEW

**Comment by eoger** [comment]
@dp-mason do you mind rebasing against `master`? There's a conflict in `CHANGES_UNRELEASED.md` preventing the merge. Also you will need to [sign your commits](https://help.github.com/en/github/authenticating-to-github/signing-commits).

File: CHANGES_UNRELEASED.md [hunk 1]
```diff
@@ -2,4 +2,24 @@
 
 # Unreleased Changes
 
+<<<<<<< HEAD
```
Comment (thomcc) [inline CHANGES_UNRELEASED.md]: Sorry, looks like you need to resolve this conflict still (Let us know if you need help with this)

**Comment by eoger** [comment]
Also it appears you rebased on top of @thomcc commit and amended it: "Set LTO to thin. Fixes #2531. [ci full]". I would suggest reading a little bit about git merging and [rebasing](https://git-scm.com/book/en/v2/Git-Branching-Rebasing) strategies.

**Comment by vladikoff** [comment]
rebased in https://github.com/mozilla/application-services/pull/2822

**Pull request closed without merging**